File: app/src/test/java/net/tlalka/android/fiszki/models/entities/LessonTest.java
package net.tlalka.android.fiszki.models.entities;

import net.tlalka.android.fiszki.models.types.LanguageType;
import net.tlalka.android.fiszki.models.types.LevelType;
import net.tlalka.android.fiszki.models.types.OwnerType;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class LessonTest {

    @Test
    public void testCheekDefaultConstructor() {

        // when
        Lesson lesson = new Lesson();

        // then
        assertNotNull(lesson);
        assertEquals(0, lesson.getId());
        assertEquals(0, lesson.getProgress());
        assertEquals(0, lesson.getScore());
    }

    @Test
    public void testCheekCustomConstructor() {

        // given
        String name = "Colors";
        LevelType levelType = LevelType.ADVANCED;
        LanguageType languageType = LanguageType.EN;

        // when
        Lesson lesson = new Lesson(name, levelType, languageType);

        // then
        assertNotNull(lesson);
        assertEquals(name, lesson.getName());
        assertEquals(levelType, lesson.getLevelType());
        assertEquals(languageType, lesson.getLanguageType());
        assertEquals(OwnerType.SYSTEM, lesson.getOwnerType());
    }
}

File: app/src/test/kotlin/net/tlalka/fiszki/model/entities/LessonTest.kt
package net.tlalka.fiszki.model.entities

import net.tlalka.fiszki.model.entities.Lesson
import net.tlalka.fiszki.model.types.LanguageType
import net.tlalka.fiszki.model.types.LevelType
import net.tlalka.fiszki.model.types.OwnerType
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Test

class LessonTest {

    @Test
    fun testCheekDefaultConstructor() {

        // when
        val lesson = Lesson()

        // then
        assertNotNull(lesson)
        assertEquals(0, lesson.id)
        assertEquals(0, lesson.progress.toLong())
        assertEquals(0, lesson.score.toLong())
    }

    @Test
    fun testCheekCustomConstructor() {

        // given
        val name = "Colors"
        val levelType = LevelType.ADVANCED
        val languageType = LanguageType.EN

        // when
        val lesson = Lesson(name, levelType, languageType)

        // then
        assertNotNull(lesson)
        assertEquals(name, lesson.name)
        assertEquals(levelType, lesson.levelType)
        assertEquals(languageType, lesson.languageType)
        assertEquals(OwnerType.SYSTEM, lesson.ownerType)
    }
}
